packages/core/src/state/selectors/useDenormalized.ts

```diff
@@ -74,31 +78,47 @@ export default function useDenormalized<
       }
     }
 
-    // inferred results are considered stale
-    if (options && options.invalidIfStale && !cacheResults) entities = {};
-
     if (params && !denormalizeCache.results[getFetchKey(params)])
       denormalizeCache.results[getFetchKey(params)] = new WeakListMap();
 
     // second argument is false if any entities are missing
     // eslint-disable-next-line prefer-const
-    return denormalize(
+    const [value, found, deleted, resolvedEntities] = denormalize(
       results,
       schema,
       entities,
       denormalizeCache.entities,
       params ? denormalizeCache.results[getFetchKey(params)] : undefined,
-    ) as [DenormalizeNullable<Shape['schema']>, boolean, boolean];
+    ) as [
+      DenormalizeNullable<Shape['schema']>,
+      boolean,
+      boolean,
+      Record<string, Record<string, any>>,
+    ];
+
+    // oldest entity dictates age
+    let expiresAt = Infinity;
+    if (found) {
+      // using Object.keys ensures we don't hit `toString` type members
+      for (const key of Object.keys(resolvedEntities)) {
+        for (const pk of Object.keys(resolvedEntities[key])) {
+          expiresAt = Math.min(expiresAt, entityMeta?.[key]?.[pk].expiresAt);
+        }
+      }
```

Comment:
you'll get a bit more perf in some engines by using `.forEach` here rather than for..of loops, which invokes the iterator protocol.

Reply:
TIL

Reply:
This is the total opposite of python